### PR TITLE
issue #10140: Validate ECJ options on ecj release

### PR DESCRIPTION
**Pull request by pbludov**
Issue #10140

New compiler options: None
Removed compiler options:
* org.eclipse.jdt.core.compiler.annotation.inheritNullAnnotations
* org.eclipse.jdt.core.compiler.problem.unsafeTypeOperation
* org.eclipse.jdt.core.compiler.annotation.missingNonNullByDefaultAnnotation
* org.eclipse.jdt.core.compiler.annotation.nullanalysis

### CODE REVIEW

**Comment by rnveach** [comment]
Is there any additions to the static analyzer in eclipse that we can use?

**Comment by pbludov** [comment]
> Is there any additions to the static analyzer in eclipse that we can use?

```bash
$ grep -o "^[^=#]*" config/org.eclipse.jdt.core.prefs | sort > /tmp/cs-options
$ curl -s https://raw.githubusercontent.com/eclipse/eclipse.jdt.core/master/org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java | grep -o 'org.eclipse.jdt.core.compiler.problem.[^"]*' | sort > /tmp/all-options
$ diff --unified /tmp/cs-options /tmp/all-options
--- /tmp/cs-options	2021-06-19 11:01:24.524094756 +0800
+++ /tmp/all-options	2021-06-19 11:00:03.940483748 +0800
@@ -1,6 +1,3 @@
-org.eclipse.jdt.core.compiler.annotation.inheritNullAnnotations
-org.eclipse.jdt.core.compiler.annotation.missingNonNullByDefaultAnnotation
-org.eclipse.jdt.core.compiler.annotation.nullanalysis
 org.eclipse.jdt.core.compiler.problem.annotatedTypeArgumentToUnannotated
 org.eclipse.jdt.core.compiler.problem.annotationSuperInterface
 org.eclipse.jdt.core.compiler.problem.APILeak
@@ -97,7 +94,6 @@
 org.eclipse.jdt.core.compiler.problem.unnecessaryElse
 org.eclipse.jdt.core.compiler.problem.unnecessaryTypeCheck
 org.eclipse.jdt.core.compiler.problem.unqualifiedFieldAccess
-org.eclipse.jdt.core.compiler.problem.unsafeTypeOperation
 org.eclipse.jdt.core.compiler.problem.unstableAutoModuleName
 org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownException
 org.eclipse.jdt.core.compiler.problem.unusedDeclaredThrownExceptionExemptExceptionAndThrowable
```

File: .ci/eclipse-compiler-javac.sh [hunk 1]
```diff
@@ -9,8 +9,11 @@ fi
 
 JAVA_RELEASE=${2:-1.8}
 
-ECJ_JAR="ecj-4.17.jar"
-ECJ_MAVEN_VERSION="R-4.17-202009021800"
+# Eclipse releases every 13 weeks: https://wiki.eclipse.org/SimRel/Simultaneous_Release_Cycle_FAQ
+# After that these variables should be updated.
+ECJ_RELEASE=R4_20
```
Comment (pbludov) [inline .ci/eclipse-compiler-javac.sh:14]: maybe there is a way to resolve `latest` to something like `4.20` and then infer all there variables?

**Review by romani — APPROVED** [review]
please rebase, pushed to master changes to fix semaphore

**Review by strkkk — APPROVED** [review]

**Review by rnveach — APPROVED** [review]

**Pull request merged**